include/inexor/vulkan-renderer/world/cube.hpp

```diff
@@ -56,13 +60,17 @@ class Cube : public std::enable_shared_from_this<Cube> {
     };
 
 private:
-    Type m_type{Type::SOLID};
+    static constexpr Type DEFAULT_TYPE = Type::EMPTY;
```

Comment:
I think this is currently used only once. I think this can be removed.

Reply:
Another note using `EMPTY` as default type, may not be that good, as it might be optimized away in the future. E.g. all sub cubes from an octant are empty => set type from octant to empty or if a world (aka. root cube) is empty remove it at all.

Reply:
I think all the points mentioned here come down to personal preference. (And obviously I implemented my preferences.)
@yeetari, @IAmNotHanni, I would love to hear your opinions on that and then I'd implement the most preferred version.

> e.g. all sub cubes from an octant are empty => set type from octant to empty

Same stands for `SOLID`.

> if a world (aka. root cube) is empty remove it at all.

I think that shouldn't happen. Also, when creating a new world, it would (in my eyes) by beneficial to create an empty cube and not automatically fill everything with a solid cube.
Lastly, I think most people who edit, build the world in a "constructive" way: They have an empty cube and fill it with subcubes. Now, when changing the type of a cube from `EMPTY` to `OCTANT`, all of the sub-cubes except for one become `EMPTY` again.

(However, the opposite stands when building in a "destructive" way, where someone removes cubes from octants. So it again comes down to personal preference.)